Incompatibility of new default host_arch x86 with arch amd64

Hi Sean,

The default option for host_arch:
https://github.com/seanmiddleditch/gha-setup-vsdevenv/commit/907c08c76ad1e7dbb81a0592b36d55d7ebae6743#diff-090b05cdc4c33b5b4d778c13de44d8eaR68
seems to be incompatible with the default architecture amd64.

We had a GitHub action set by mistake to follow branch master and got these errors:
```
 LINK: command "C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1426~1.288\bin\Hostx86\x86\link.exe /nologo CMakeFiles\cmTC_92de8.dir\testCXXCompiler.cxx.obj /out:cmTC_92de8.exe /implib:cmTC_92de8.lib /pdb:cmTC_92de8.pdb /version:0.0 /machine:X86 /debug /subsystem:console kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:cmTC_92de8.exe.manifest" failed (exit code 1120) with the following output:
    testCXXCompiler.cxx.obj : error LNK2001: unresolved external symbol __RTC_InitBase
    testCXXCompiler.cxx.obj : error LNK2001: unresolved external symbol __RTC_Shutdown
    LINK : error LNK2001: unresolved external symbol _mainCRTStartup
    C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64\kernel32.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'x86'
    C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64\user32.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'x86'
    C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64\gdi32.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'x86'
    C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64\winspool.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'x86'
   ...
```
Our workaround was to go back to tag v1, but I thought you may want to know.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -19,6 +19,6 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-node@v1
       - uses: ./
-      - run: cl /EHsc /Fe:test.exe tests/test.cpp && test.exe
+      - run: 'cl /EHsc /Fe:test.exe tests\\test.cpp && .\\test.exe'
         if: runner.os == 'Windows'
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -19,5 +19,7 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-node@v1
       - uses: ./
-      - run: cl.exe
+      - run: cl /EHsc tests/test.cpp && tests/test.exe
+        shell: bash
         if: runner.os == 'Windows'
+

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -19,7 +19,6 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-node@v1
       - uses: ./
-      - run: cl /EHsc tests/test.cpp && tests/test.exe
-        shell: bash
+      - run: cl /EHsc /o test.exe tests/test.cpp && test.exe
         if: runner.os == 'Windows'
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -19,6 +19,6 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-node@v1
       - uses: ./
-      - run: cl /EHsc /o test.exe tests/test.cpp && test.exe
+      - run: cl /EHsc /Fe:test.exe tests/test.cpp && test.exe
         if: runner.os == 'Windows'
 

--- a/README.md
+++ b/README.md
@@ -18,7 +18,7 @@ Inputs
 
 - `vswhere`: Path to `vswhere.exe` (default system-installed copy).
 - `arch`: Build architecture (default `amd64`).
-- `host_arch`: Host architecture (default `x86`).
+- `host_arch`: Host architecture (default `x86_amd64`).
 - `winsdk`: WinSDK version override.
 - `toolset_version`: Build toolset version override.
 - `components`: List of required VS components, semi-colon separated.

--- a/README.md
+++ b/README.md
@@ -18,7 +18,7 @@ Inputs
 
 - `vswhere`: Path to `vswhere.exe` (default system-installed copy).
 - `arch`: Build architecture (default `amd64`).
-- `host_arch`: Host architecture (default `x86_amd64`).
+- `host_arch`: Host architecture override.
 - `winsdk`: WinSDK version override.
 - `toolset_version`: Build toolset version override.
 - `components`: List of required VS components, semi-colon separated.

--- a/action.yml
+++ b/action.yml
@@ -11,8 +11,8 @@ inputs:
     description: 'Target build architecture'
     default: 'amd64'
   host_arch:
-    description: 'Host architecture (x86 or amd64)'
-    default: 'x86'
+    description: 'Host architecture (x86, x86_amd64, or amd64)'
+    default: 'x86_amd64'
   winsdk:
     description: 'Overrides the default WinSDK version'
     default: ''

--- a/action.yml
+++ b/action.yml
@@ -12,7 +12,7 @@ inputs:
     default: 'amd64'
   host_arch:
     description: 'Host architecture (x86, x86_amd64, or amd64)'
-    default: 'x86_amd64'
+    default: ''
   winsdk:
     description: 'Overrides the default WinSDK version'
     default: ''

--- a/index.js
+++ b/index.js
@@ -10,7 +10,7 @@ try {
     }
 
     const arch = core.getInput('arch') || 'amd64'
-    const hostArch = core.getInput('host_arch') || 'x86_amd64'
+    const hostArch = core.getInput('host_arch') || ''
     const toolsetVersion = core.getInput('toolset_version') || ''
     const winsdk = core.getInput('winsdk') || ''
     const vswhere = core.getInput('vswhere') || 'vswhere.exe'
@@ -33,6 +33,8 @@ try {
         '-property', 'installationPath',
     ].concat(requiresArg)
 
+    console.log(`$ ${vswherePath} ${vswhereArgs.join(' ')}`)
+
     const vswhereResult = spawn(vswherePath, vswhereArgs, {encoding: 'utf8'})
     if (vswhereResult.error) throw vswhereResult.error
     const installPathList = vswhereResult.output.filter(s => !!s).map(s => s.trim())
@@ -45,17 +47,17 @@ try {
     const vsDevCmdPath = path.win32.join(installPath, 'Common7', 'Tools', 'vsdevcmd.bat')
     console.log(`vsdevcmd: ${vsDevCmdPath}`)
 
-    const vsDevCmdArgs = [
-        vsDevCmdPath,
-        `-arch=${arch}`,
-        `-host_arch=${hostArch}`
-    ]
+    const vsDevCmdArgs = [ vsDevCmdPath, `-arch=${arch}` ]
+    if (hostArch != '')
+        vsDevCmdArgs.push(`-host_arch=${hostArch}`)
     if (toolsetVersion != '')
         vsDevCmdArgs.push(`-vcvars_vers=${toolsetVersion}`)
     if (winsdk != '')
         vsDevCmdArgs.push(`-winsdk=${winsdk}`)
     
     const cmdArgs = [ '/q', '/k'].concat(vsDevCmdArgs, ['&&', 'set'])
+
+    console.log(`$ cmd ${cmdArgs.join(' ')}`)
 
     const cmdResult = spawn('cmd', cmdArgs, {encoding: 'utf8'})
     if (cmdResult.error) throw cmdResult.error

--- a/index.js
+++ b/index.js
@@ -10,7 +10,7 @@ try {
     }
 
     const arch = core.getInput('arch') || 'amd64'
-    const hostArch = core.getInput('host_arch') || 'x86'
+    const hostArch = core.getInput('host_arch') || 'x86_amd64'
     const toolsetVersion = core.getInput('toolset_version') || ''
     const winsdk = core.getInput('winsdk') || ''
     const vswhere = core.getInput('vswhere') || 'vswhere.exe'

--- a/tests/test.cpp
+++ b/tests/test.cpp
@@ -1,0 +1,5 @@
+#include <iostream>
+
+int main() {
+    std::cout << "true\n";
+}

